x/rollapp/keeper/invariants.go

```diff
@@ -0,0 +1,204 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+)
+
+// RegisterInvariants registers the bank module invariants
+func RegisterInvariants(ir sdk.InvariantRegistry, k Keeper) {
+	ir.RegisterRoute(types.ModuleName, "rollapp-state-index", RollappLatestStateIndexInvariant(k))
+	ir.RegisterRoute(types.ModuleName, "rollapp-count", RollappCountInvariant(k))
+	ir.RegisterRoute(types.ModuleName, "block-height-to-finalization-queue", BlockHeightToFinalizationQueueInvariant(k))
+	ir.RegisterRoute(types.ModuleName, "rollapp-by-eip155-key", RollappByEIP155KeyInvariant(k))
+}
+
+// AllInvariants runs all invariants of the X/bank module.
+func AllInvariants(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		res, stop := RollappLatestStateIndexInvariant(k)(ctx)
+		if stop {
+			return res, stop
+		}
+		res, stop = RollappCountInvariant(k)(ctx)
+		if stop {
+			return res, stop
+		}
+		res, stop = BlockHeightToFinalizationQueueInvariant(k)(ctx)
+		if stop {
+			return res, stop
+		}
+		res, stop = RollappByEIP155KeyInvariant(k)(ctx)
+		if stop {
+			return res, stop
+		}
+		return "", false
+	}
+}
+
+func RollappByEIP155KeyInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var (
+			broken bool
+			msg    string
+		)
+
+		rollapps := k.GetAllRollapp(ctx)
+		for _, rollapp := range rollapps {
+			eip155, err := types.ParseChainID(rollapp.RollappId)
+			if err != nil {
+				msg += fmt.Sprintf("rollapp (%s) have invalid rollappId\n", rollapp.RollappId)
+				broken = true
+				continue
+			}
+
+			//not breaking invariant, as eip155 format is not required
+			if eip155 == nil {
+				continue
+			}
+
+			_, found := k.GetRollappByEIP155(ctx, eip155.Uint64())
+			if !found {
+				msg += fmt.Sprintf("rollapp (%s) have no eip155 key\n", rollapp.RollappId)
+				broken = true
+				continue
+			}
+		}
+
+		return sdk.FormatInvariant(
+			types.ModuleName, "rollapp-by-eip155-key",
+			msg,
+		), broken
+	}
+}
+
+func BlockHeightToFinalizationQueueInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var (
+			broken bool
+			msg    string
+		)
+
+		for _, rollapp := range k.GetAllRollapp(ctx) {
+			if !k.IsRollappStarted(ctx, rollapp.RollappId) {
+				continue
+			}
+			latestStateIdx, _ := k.GetLatestStateInfoIndex(ctx, rollapp.RollappId)
+			latestFinalizedStateIdx, _ := k.GetLatestFinalizedStateIndex(ctx, rollapp.RollappId)
```

Comment:
what if no states where yet finalized?

Reply:
it will be `0` and the `firstUnfinalizedStateIdx` will be `1` as expected